### PR TITLE
SDK-100 follow-up fixes

### PR DESCRIPTION
**Pull request by pgutkowski**
I have separated integration tests as @rkorytkowski suggested in #40. With this packaging structure, plugin is built first, then integration tests are conducted.
I'm not sure if naming and POMs are 100% correct, I would be thankful for any tips.

### CODE REVIEW

File: openmrs-sdk-maven-plugin-integration-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.openmrs.maven.plugins</groupId>
+        <artifactId>openmrs-sdk-maven-plugin-main</artifactId>
+        <version>2.1.3-SNAPSHOT</version>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin-integration-test/pom.xml]: Let's make the parent org.openmrs.maven:openmrs-sdk:2.1.3-SNAPSHOT. The reasoning is that we will want to include different artifacts (not only plugins, but also archetypes) within the sdk.

File: openmrs-sdk-maven-plugin-integration-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    </parent>
+
+    <groupId>org.openmrs.maven.plugins</groupId>
+    <artifactId>openmrs-sdk-maven-plugin-integration-test</artifactId>
+    <version>2.1.3-SNAPSHOT</version>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin-integration-test/pom.xml]: Let's make it org.openmrs.maven:openmrs-sdk-integration-tests:2.1.3-SNAPSHOT

File: openmrs-sdk-maven-plugin-integration-test/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            <organization>OpenMRS</organization>
+            <organizationUrl>http://openmrs.org/</organizationUrl>
+        </developer>
+        <developer>
+            <name>Dmytro Trifonov</name>
+            <email>dmytro.trifonov@openmrs.org</email>
+            <organization>OpenMRS</organization>
+            <organizationUrl>http://openmrs.org/</organizationUrl>
+        </developer>
+    </developers>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin-integration-test/pom.xml]: It's enough to have licences and developers in main pom.

File: openmrs-sdk-maven-plugin-integration-test/pom.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    <scm>
+        <connection>scm:git:https://github.com/openmrs/openmrs-contrib-sdk-maven-plugin</connection>
+        <developerConnection>scm:git:https://github.com/openmrs/openmrs-contrib-sdk-maven-plugin</developerConnection>
+        <url>https://github.com/openmrs/openmrs-contrib-sdk-maven-plugin</url>
+        <tag>HEAD</tag>
+    </scm>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin-integration-test/pom.xml]: scm in main pom only.

File: openmrs-sdk-maven-plugin-integration-test/pom.xml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <tomcat.version>7.0.67</tomcat.version>
+    </properties>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin-integration-test/pom.xml]: properties in main pom only

File: openmrs-sdk-maven-plugin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        <url>https://github.com/openmrs/openmrs-contrib-sdk-maven-plugin</url>
+        <tag>HEAD</tag>
+    </scm>
+
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <tomcat.version>7.0.67</tomcat.version>
+    </properties>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin/pom.xml]: licenses, developers, scm and properties are not needed here. They should be inherited from main pom.

File: openmrs-sdk-maven-plugin/pom.xml [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </snapshotRepository>
+        <repository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
```
Comment (rkorytkowski) [inline openmrs-sdk-maven-plugin/pom.xml]: Repositories and distributionManagement should be in main pom only.

File: pom.xml [hunk 8]
```diff
@@ -1,338 +1,96 @@
...
+            <email>rafal@openmrs.org</email>
+            <organization>OpenMRS</organization>
+            <organizationUrl>http://openmrs.org/</organizationUrl>
+        </developer>
+        <developer>
+            <name>Dmytro Trifonov</name>
+            <email>dmytro.trifonov@openmrs.org</email>
+            <organization>OpenMRS</organization>
+            <organizationUrl>http://openmrs.org/</organizationUrl>
+        </developer>
```
Comment (rkorytkowski) [inline pom.xml:39]: You can add yourself here :)

File: pom.xml [hunk 9]
```diff
@@ -1,338 +1,96 @@
...
+            <version>1.3</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
+            <version>1.3</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (rkorytkowski) [inline pom.xml]: It's not recommended to specify dependencies in main pom. If you want to share the version then put them in dependencyManagement section in the main pom.

File: pom.xml [hunk 10]
```diff
@@ -1,338 +1,96 @@
...
+    <packaging>pom</packaging>
+    <version>2.1.3-SNAPSHOT</version>
+
+    <name>OpenMRS SDK Maven Plugin</name>
+    <description>OpenMRS SDK allows for rapid development of OpenMRS modules.</description>
+    <url>https://wiki.openmrs.org/display/docs/OpenMRS+SDK</url>
+
+    <modules>
+        <module>openmrs-sdk-maven-plugin</module>
+        <module>openmrs-sdk-maven-plugin-integration-test</module>
```
Comment (rkorytkowski) [inline pom.xml]: rename to integration-tests

**Comment by pgutkowski** [comment]
@rkorytkowski fixes done!

**Pull request merged**